subprojects/ivy/src/integTest/groovy/org/gradle/api/publish/ivy/IvyPublishJavaIntegTest.groovy

```diff
@@ -699,6 +699,51 @@ class IvyPublishJavaIntegTest extends AbstractIvyPublishIntegTest {
         }
     }
 
+    def "can publish java component with capabilities"() {
+        given:
+        createBuildScripts("""
+            dependencies {
+                capabilities {
+                    capability('org:cap') {
+                        providedBy 'org:foo'
+                    }
```

Comment:
It seems like there should be an easy way to define a capability that the published component provides. Or is this implicit? 

If it's implicit in a locally-defined capability is provided by the declaring component, then we need to either a) map this explicitly to the published metadata or b) treat published components as always implicitly providing all components they declare.

If it's not implicit, then I think we need a convenience to make it clear.

Reply:
There's nothing implicit. I think there would be a lot of performance problems if we implicitly added a capability for each published module. But there's no "easy way" to declare a capability for the same module either. I don't think it's necessary for the first take, we can always add that later, if we feel it's too much overhead to declare it in the `dependencies` block. But I'd rather see how those things are used before adding a shortcut.